includes/wc-template-functions.php

```diff
@@ -301,7 +301,13 @@ function wc_generator_tag( $gen, $type ) {
 function wc_body_class( $classes ) {
 	$classes = (array) $classes;
 
-	if ( is_woocommerce() ) {
+	if ( is_shop() ) {
+
+		$classes[] = 'woocommerce';
+		$classes[] = 'woocommerce-shop';
+		$classes[] = 'woocommerce-page';
+
+	} elseif ( is_woocommerce() ) {
```

Comment:
The `is_woocommerce` function triggers a filter with the same name, if it returns `false` then the `woocommerce` and `woocommerce-page` classes aren't supposed to be applied. Therefore, I suggest the following instead:

```suggestion
	if ( is_shop() ) {

		$classes[] = 'woocommerce-shop';

	} 
	
	if ( is_woocommerce() ) {
```

Reply:
Thanks for clarifying this, @Konamiman. 🙌